index.js

```diff
@@ -3,9 +3,26 @@ const fs = require('fs');
 const path = require('path');
 const uniqueString = require('unique-string');
 const tempDir = require('temp-dir');
+const isStream = require('is-stream');
+const pify = require('pify');
+const writeFile = pify(fs.writeFile);
 
 const getPath = () => path.join(tempDir, uniqueString());
 
+const writeStream = async (filePath, fileContent) => new Promise((resolve, reject) => {
+	const writable = fs.createWriteStream(filePath);
+
+	fileContent
```

Comment:
Use `stream.pipeline` instead of `.pipe`. It handles errors better. Read up on how it works.

Reply:
Fixed!